public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -97,6 +111,69 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
 
   private forceRouteReloadCounter = 0;
   state: State = this.getCleanState();
+  maxFileSize = 500000;
+
+  onFileDrop = (acceptedFiles: File[], fileRejections: FileRejection[], event: DropEvent) => {
+    console.log(acceptedFiles);
+    acceptedFiles.forEach((file) => {
+      const reader = new FileReader();
+      reader.readAsArrayBuffer(file);
+      reader.onload = () => {
+        const result = reader.result;
+        if (result && result instanceof ArrayBuffer) {
+          const dataFrames = readSpreadsheet(result);
+          const snapshot: DataFrameJSON[] = [];
+          dataFrames.forEach((df) => {
+            const dataframeJson = dataFrameToJSON(df);
+            snapshot.push(dataframeJson);
+          });
+          const grafanaDS = {
+            type: 'grafana',
+            uid: 'grafana',
+          };
+          this.props.dashboard?.addPanel({
```

Comment:
Seems like the creation of this literal should be extracted into a function or similar. Seems a little bit awkward this way

Reply:
I agree. IDK if it's possible to get this from the plugin.json for the grafana ds in any way. Would be nice.